public/app/features/scenes/scenes/demo.tsx

```diff
@@ -94,20 +116,28 @@ export function getScenePanelRepeaterTest(): Scene {
           new SceneFlexLayout({
             size: { minHeight: 200 },
             children: [
-              new VizPanel({
-                pluginId: 'timeseries',
-                title: 'Title',
-                options: {
-                  legend: { displayMode: 'hidden' },
-                },
+              new SceneFlexChild({
+                children: [
+                  new VizPanel({
+                    pluginId: 'timeseries',
+                    title: 'Title',
+                    options: {
+                      legend: { displayMode: 'hidden' },
+                    },
+                  }),
+                ],
               }),
-              new VizPanel({
-                size: { width: 300 },
-                pluginId: 'stat',
-                fieldConfig: { defaults: { displayName: 'Last' }, overrides: [] },
-                options: {
-                  graphMode: 'none',
-                },
+              new SceneFlexChild({
+                children: [
+                  new VizPanel({
+                    size: { width: 300 },
```

Comment:
@torkelo I don't have a super strong opinion here, but my feeling is the separation makes sense, at least at this stage when we are trying to find good interfaces. My main point for the separation is that the sizing for Flex and for Grid are two completely different things - at least in the current implementation where i.e. `width?: number | string;
  height?: number | string; ` are quite open-ended.  

Handling both grid and flex sizing via the same interface will introduce complexity in the sizing/position calculation in the consuming components. Unless we agree that we use the grid units for the Flex layout too - just thought about it, maybe something worth exploring more?

Reply:
yea, I super torn, think both directions have strong pros & cons. The biggest con is just the verbosity of the "as code" story and the nesting. I want the as code story as simple as possible, with defaults that just work (which is why size properly is optional). 

But thinking from the "as code" perspective I think even in the grid layout we could have some lax requirements where cells don't need any size or pposition, we can position things automatically if they lack position and dimensions (using some simple heuristic). And grid cells could say "width: 50%" or so and the layout system will translate that into grid sizes when it builds layout for the grid. 

Regarding if flex layout children should use grid coords, that sounds like a good idea. should make it easier to have things align and be same widths.

Reply:
The more I think about it, the more I strive to get things back to size property on SceneObjects, rather than specialized layout item components- this would resemble how Photoshop/Figma/etc treat objects in a canvas, where sizing is basically a common property of any kind of object, without requiring any other, higher level container to be added in order to provide sizing and positioning. While having those layout item containers separated looks elegant from as-code pov, I have a feeling it will give us more headaches going deeper into building scenes via UI.

Thinking about as-code I can see one risk here though, which is related to reusable panels/objects. If the panel definition is considered a single, reusable unit that can be used in different layouts/scenes, then the size property being part of must be considered the _default_ size of an object that can be overridden, rather than a fixed property that's just given. This means basically that serialized reference to an object is no longer a plain 

```ts
{
  $ref: 'https://GitHub link to a panel for example'
}
```

but also can contain overridable properties:

```ts
{
  $ref: 'https://GitHub link to a panel for example'
  size: { ... }
}
```

Reply:
> would resemble how Photoshop/Figma/etc treat objects in a canvas, where sizing is basically a common property of any kind of object,

Yes, this was my inspiration as well for a common size property (name is hard for this property) 

> Thinking about as-code I can see one risk here though, which is related to reusable panels/objects. If the panel definition is considered a single, reusable unit that can be used in different layouts/scenes, then the size property being part of must be considered the default size of an object that can be overridden, rather than a fixed property that's just given. This means basically that serialized reference to an object is no longer a plain

I had the exact same thought/consideration as well but I came to the conclusion that we will need this default / override behavior anyway. Think of a panel title for example, that is overridable from a library panel reference. The same goes for panel options and field config and data queries. They are on the same level today (VizPanel state). 

To make references completely isolated we would have to segregate visualization options (panel plugin id, options, fieldOptions), queries (queryOptions + queries), panel options (title, description, etc), size/grid props

```ts
 {
   visualization: { pluginId: 'timeseries', options: {}, fieldConfig: {} }, 
   panelOptions: { title, description, repeat },  // This could also be flat directly on VizPanel 
   $data: { options, queries },
   $size: { }, 
}
```

```ts
 {
   visualization: { ref: 'some id for a standalone reusable saved viz' }, 
   panelOptions: { title, description, repeat },  // This could also be flat directly on VizPanel 
   $data: {  options, queries [ { ref: 'id for query object' }], 
   $size: { }, 
}
```

A libary panel that represents both data (queries) and visualization settings but not panel title and panel position/size is harder to represent without override ability.

Reply:
@torkelo I got rid of the layout child components (Flex/GridChild) - I think it to have the size a common property on renderable scene objects rather that adding this additional layer just to handle the size. In the end those objects' responsibility is just wrapping children in a DOM element, that can easily be done within the layout.